Kickstarter-iOS/Views/UIView+ShimmerLoading.swift

```diff
@@ -0,0 +1,174 @@
+import Foundation
+import Library
+import ObjectiveC
+import UIKit
+
+private let gradientLayerName = "ksr_shimmer_gradientLayer"
+
+private enum ShimmerConstants {
+  enum Locations {
+    static let start: [NSNumber] = [-1.0, -0.5, 0.0]
+    static let end: [NSNumber] = [1.0, 1.5, 2.0]
+  }
+
+  enum Animation {
+    static let movingAnimationDuration: CFTimeInterval = 1.25
+    static let delayBetweenAnimationLoops: CFTimeInterval = 0.3
+  }
+}
+
+private struct AssociatedKeys {
+  static var isLoading = "isLoading"
+  static var shimmersWhenLoading = "shimmersWhenLoading"
+  static var shimmerLayers = "shimmerLayers"
+}
+
+protocol ShimmerLoading: AnyObject {
+  func startLoading()
+  func stopLoading()
+}
+
+extension UIView {
+  var shimmersWhenLoading: Bool {
+    get {
+      guard let value = objc_getAssociatedObject(self, &AssociatedKeys.shimmersWhenLoading) as? Bool else {
+        return false
+      }
+      return value
+    }
+    set(newValue) {
+      objc_setAssociatedObject(
+        self,
+        &AssociatedKeys.shimmersWhenLoading,
+        newValue,
+        objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC
+      )
+    }
+  }
+}
+
+extension ShimmerLoading where Self: UIView {
+  private var isLoading: Bool {
```

Comment:
Is this private `isLoading` property really needed, or can we just pass it into `updateAnimation` directly?

```
func startLoading() {
    self.updateAnimation(isLoading: true)
}
```